[CI] ToDoubleTests testFold {TestCase=<long unicode KEYWORD>} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/hha6322mmmo2w/tests/:x-pack:plugin:esql:test/org.elasticsearch.xpack.esql.expression.function.scalar.convert.ToDoubleTests/testFold%20%7BTestCase=%3Clong%20unicode%20KEYWORD%3E%7D


**Reproduction line:**
```
gradlew ':x-pack:plugin:esql:test' --tests "org.elasticsearch.xpack.esql.expression.function.scalar.convert.ToDoubleTests.testFold {TestCase=<long unicode KEYWORD>}" -Dtests.seed=78A9DE0D83EC6C83 -Dtests.locale=hr-HR -Dtests.timezone=Europe/Nicosia -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.expression.function.scalar.convert.ToDoubleTests#testFold {TestCase=<long unicode KEYWORD>}`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testFold%20%7BTestCase%3D%3Clong%20unicode%20KEYWORD%3E%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.expression.function.scalar.convert.ToDoubleTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a collection containing <WARN: Line -1:-1: java.lang.NumberFormatException: For input string: \"%05Pxr%1F\"w7%158M*%0C%1Du%7F%08%17C6Mwlk%00M|%1F-%06c%00!%08%11UlTU_$*'ueVv4e%03&#%0D;t\"bI%7F_?%18UFY%05di$%18D%1A%0E>c`<%03-%10%0F%10k%02X0I%18C2E%06x4J6%15)%7FQ0%04b!`%07%1FPS_%25o#y}%0Dd}W%0Ae7f%16%1B`%0C%00kX%16%0B%08%14`Us%14%1D%1Dv%1E%0B%0C()%0CiB%06a^5c%1C%05YR0.%25]z<%1AU*ng%05k3q1%06	=c~iNB90Uy%25<N%15^jT%0AR|w}I7%1C%12%0F\"vn\">:%0C;;z%03%06i/y%1FbyuTW%19\"KyKA%04%1A%1D8%19^Pt%08>r2+/O-4%15%17IIpvO^/T 84%0D%1A{%08#i2%13E#yc %16h]?<|7^%04%0FFT%01'E%1F{%14N\"=kja;OK\\@%1B5\\\\H;,a	&<%0E9m%1Ew`:^%12z[%0C_(%7F9!%11R)3{WTZ?KL3e}%10%17*l%06@kN%17EA0-7zJx%01\"Na%18:%14u[%1D%13|dF+~]e%175D%00V[X`%0FNYNF%0B%12%009_	j%1E%1C4C!b{%0BkA|*\\&rx,iE8'3l]95g%16aSh'ewzKHw4d[:O|wKM!rh	%0D#;1)YM$ \"%123x`'7%0B%06s8%05|%01%05@JB%1F	ZC2F[L%08f%0D^%03&qmlMcG%173t\\2TKMO@l~|H9lKq\\M{%02NC%1C;%0C$W%15Ub'%0CCInXF=3['nCH%7F%00%0F!%04L%12j%25/?3l%1AQA%15T%04&-(6n\"#%1Ef6Wc.\\]__x%08.\\%14B%15Y:@/k'(S4SIt%7FV9%0B&VL2t%051{m	s8}%1A%0F%0AR%15%04SCm%15?;dZb}%16`Bq\"q>(TB7n%1BK8cj%03>)b9MF!%05%00K2%16F%10\\^.\"'C%17Q,%12O%10%00C|=0cw6G%18b'<u]ZR;%17%04A}w$M,p&%16%1A%056%15i>%1EWl%08J~zk%0D%0A:k%25%19%0C%01]%0C9<R%05pO%1F%1C%06>%1E9%19<%01O{%17NX?Kh%13hIQjX]%05#%14%1CXp|%11%06%0D%1DZN%04)1%0B%05S\\%11!@tz{xH%15[g%1B+6?%0Cp%1D%1A%19%1E%17W*++K^%1Cp^U%13%0B}%15>%06%15%05,)6J%029.%19\"qA'M#cz&54c$'a3}veG-N8%0BwGI%06;%197& %14+b%160&n*2%04-$%16	1*r-OzoK%10%02Q`\".z%7F%1BN!&@yxh&e_%0DzEK%15%7F%00)%19z.l/%17U,;%00{[%1D%01zW>8%25;4y{Q!%1E^/}%02AAXHa%0B:0qn	%0C!@3%19A}{d:6,sH.~R\\y%00u0pI/1)a%0D@%0F6z+%02J qPr3O%1FPh;8I%1E;hm:|R%1A5\\+G;K40~]m5]V%14z %0A~P~^tC}%03|%0D'!	!tP]%19y%04Lb%1Botm4Zs%0Ft4%1Aa8J\"+s%1EKX@%0Fk1BBOm%1E%1Dqq%1F%1AM%13L%14$^Nwa%0F	 vK%0Fk4]Bt+1%1B%064Jip%12CE%1E$k|%1Blc; %0D2%18kG4ed.%12%1E)/%0EyY:%04n&/Y%25%15q3f(db	%7FF%08%10q%0DR&~]Wq%11SOW%1Eh[G'$C]$%1Ay u%02%13X <%06LT%1BM	^H,%13%7Fx%13	Oe1V!ObBs>Do%108?Q$Fa%1Cd:NPtR%1Cg%0CRc.C%04,@5%0A%25^VdG%19-<h=3l@ol$%00ii9}%02%15%17I%0A%0EyyW%10Fg%1Dhg[%0F.VFd%0B%7F%08zY(v.(%08%0F5%0C`G0;%7Fp%0A%0A%10D,x%10%1D|4%02X$OLuo1%15~u%044%0CWoRc1Da\">
     but: mismatches were: [was <CRITICAL: Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.>, was <CRITICAL: Line -1:-1: java.lang.NumberFormatException: For input string: \"Pxr%1F\"w7%158M*%0C%1Du%7F%08%17C6Mwlk%00M|%1F-%06c%00!%08%11UlTU_$*'ueVv4e%03&#%0D;t\"bI%7F_?%18UFY%05di$%18D%1A%0E>c`<%03-%10%0F%10k%02X0I%18C2E%06x4J6%15)%7FQ0%04b!`%07%1FPS_%25o#y}%0Dd}W%0Ae7f%16%1B`%0C%00kX%16%0B%08%14`Us%14%1D%1Dv%1E%0B%0C()%0CiB%06a^5c%1C%05YR0.%25]z<%1AU*ng%05k3q1%06	=c~iNB90Uy%25<N%15^jT%0AR|w}I7%1C%12%0F\"vn\">:%0C;;z%03%06i/y%1FbyuTW%19\"KyKA%04%1A%1D8%19^Pt%08>r2+/O-4%15%17IIpvO^/T 84%0D%1A{%08#i2%13E#yc %16h]?<|7^%04%0FFT%01'E%1F{%14N\"=kja;OK\\@%1B5\\\\H;,a	&<%0E9m%1Ew`:^%12z[%0C_(%7F9!%11R)3{WTZ?KL3e}%10%17*l%06@kN%17EA0-7zJx%01\"Na%18:%14u[%1D%13|dF+~]e%175D%00V[X`%0FNYNF%0B%12%009_	j%1E%1C4C!b{%0BkA|*\\&rx,iE8'3l]95g%16aSh'ewzKHw4d[:O|wKM!rh	%0D#;1)YM$ \"%123x`'7%0B%06s8%05|%01%05@JB%1F	ZC2F[L%08f%0D^%03&qmlMcG%173t\\2TKMO@l~|H9lKq\\M{%02NC%1C;%0C$W%15Ub'%0CCInXF=3['nCH%7F%00%0F!%04L%12j%25/?3l%1AQA%15T%04&-(6n\"#%1Ef6Wc.\\]__x%08.\\%14B%15Y:@/k'(S4SIt%7FV9%0B&VL2t%051{m	s8}%1A%0F%0AR%15%04SCm%15?;dZb}%16`Bq\"q>(TB7n%1BK8cj%03>)b9MF!%05%00K2%16F%10\\^.\"'C%17Q,%12O%10%00C|=0cw6G%18b'<u]ZR;%17%04A}w$M,p&%16%1A%056%15i>%1EWl%08J~zk%0D%0A:k%25%19%0C%01]%0C9<R%05pO%1F%1C%06>%1E9%19<%01O{%17NX?Kh%13hIQjX]%05#%14%1CXp|%11%06%0D%1DZN%04)1%0B%05S\\%11!@tz{xH%15[g%1B+6?%0Cp%1D%1A%19%1E%17W*++K^%1Cp^U%13%0B}%15>%06%15%05,)6J%029.%19\"qA'M#cz&54c$'a3}veG-N8%0BwGI%06;%197& %14+b%160&n*2%04-$%16	1*r-OzoK%10%02Q`\".z%7F%1BN!&@yxh&e_%0DzEK%15%7F%00)%19z.l/%17U,;%00{[%1D%01zW>8%25;4y{Q!%1E^/}%02AAXHa%0B:0qn	%0C!@3%19A}{d:6,sH.~R\\y%00u0pI/1)a%0D@%0F6z+%02J qPr3O%1FPh;8I%1E;hm:|R%1A5\\+G;K40~]m5]V%14z %0A~P~^tC}%03|%0D'!	!tP]%19y%04Lb%1Botm4Zs%0Ft4%1Aa8J\"+s%1EKX@%0Fk1BBOm%1E%1Dqq%1F%1AM%13L%14$^Nwa%0F	 vK%0Fk4]Bt+1%1B%064Jip%12CE%1E$k|%1Blc; %0D2%18kG4ed.%12%1E)/%0EyY:%04n&/Y%25%15q3f(db	%7FF%08%10q%0DR&~]Wq%11SOW%1Eh[G'$C]$%1Ay u%02%13X <%06LT%1BM	^H,%13%7Fx%13	Oe1V!ObBs>Do%108?Q$Fa%1Cd:NPtR%1Cg%0CRc.C%04,@5%0A%25^VdG%19-<h=3l@ol$%00ii9}%02%15%17I%0A%0EyyW%10Fg%1Dhg[%0F.VFd%0B%7F%08zY(v.(%08%0F5%0C`G0;%7Fp%0A%0A%10D,x%10%1D|4%02X$OLuo1%15~u%044%0CWoRc1Da\">]

  at __randomizedtesting.SeedInfo.seed([78A9DE0D83EC6C83:F4D51799F48A951A]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.test.ESTestCase.assertWarnings(ESTestCase.java:645)
  at org.elasticsearch.test.ESTestCase.assertWarnings(ESTestCase.java:596)
  at org.elasticsearch.xpack.esql.expression.function.AbstractFunctionTestCase.testFold(AbstractFunctionTestCase.java:516)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDoubleTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDoubleTests.java
@@ -54,19 +54,13 @@ public class ToDoubleTests extends AbstractFunctionTestCase {
             List.of()
         );
         // random strings that don't look like a double
-        TestCaseSupplier.forUnaryStrings(
-            suppliers,
-            evaluatorName.apply("String"),
-            DataTypes.DOUBLE,
-            bytesRef -> null,
-            bytesRef -> {
-                var exception = expectThrows(NumberFormatException.class, () -> Double.parseDouble(bytesRef.utf8ToString()));
-                return List.of(
-                    "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
-                    "Line -1:-1: " + exception
-                );
-            }
-        );
+        TestCaseSupplier.forUnaryStrings(suppliers, evaluatorName.apply("String"), DataTypes.DOUBLE, bytesRef -> null, bytesRef -> {
+            var exception = expectThrows(NumberFormatException.class, () -> Double.parseDouble(bytesRef.utf8ToString()));
+            return List.of(
+                "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
+                "Line -1:-1: " + exception
+            );
+        });
         TestCaseSupplier.forUnaryUnsignedLong(
             suppliers,
             evaluatorName.apply("UnsignedLong"),

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDoubleTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDoubleTests.java
@@ -59,11 +59,13 @@ public class ToDoubleTests extends AbstractFunctionTestCase {
             evaluatorName.apply("String"),
             DataTypes.DOUBLE,
             bytesRef -> null,
-            bytesRef -> List.of(
-                "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
-                "Line -1:-1: java.lang.NumberFormatException: "
-                    + (bytesRef.utf8ToString().isEmpty() ? "empty String" : ("For input string: \"" + bytesRef.utf8ToString() + "\""))
-            )
+            bytesRef -> {
+                var exception = expectThrows(NumberFormatException.class, () -> Double.parseDouble(bytesRef.utf8ToString()));
+                return List.of(
+                    "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
+                    "Line -1:-1: " + exception
+                );
+            }
         );
         TestCaseSupplier.forUnaryUnsignedLong(
             suppliers,

